go/common/enclave.go

```diff
@@ -45,7 +45,7 @@ type Enclave interface {
 	// It is the responsibility of the host to gossip the returned rollup
 	// For good functioning the caller should always submit blocks ordered by height
 	// submitting a block before receiving ancestors of it, will result in it being ignored
-	SubmitL1Block(block types.Block, isLatest bool) (*BlockSubmissionResponse, error)
+	SubmitL1Block(block types.Block, isLatest bool, isSequencer bool) (*BlockSubmissionResponse, error)
```

Comment:
We're coupling rollup production to L1 block submission, to avoid a race condition where two rollups are produced at the same height if two L1 blocks come in at the same time. But we have to add this parameter, as only the sequencer should be producing blocks.